CrowServe/CrowServe.cpp

```diff
@@ -17,90 +17,61 @@ void deinit_sockets(){
 #endif
 }
 
-bool Socket::try_init() {  
+void Socket::try_init(std::stop_token stop_token) {
+    m_mutex.lock();
+    m_state = SocketState::Connecting;
+    m_mutex.unlock();
+
     addrinfo hints = {};
     addrinfo* result = nullptr;
+    addrinfo* info = nullptr;
     hints.ai_family = AF_INET;
     hints.ai_socktype = SOCK_STREAM;
     hints.ai_flags = AI_PASSIVE;
+    using namespace std::chrono_literals;
```

Comment:
I typically add `using namespace std::literals;` (which does chrono and more) somewhere in a common header, since those are generally useful everywhere.